django_project/ford3/views/provider.py

```diff
@@ -15,40 +18,79 @@
+        # upload button is pressed
+        if 'upload_logo' in request.POST:
+            form = ProviderForm(request.POST, request.FILES)
+            # users press upload button without specifying a file
+            if not request.FILES:
+                context = {
+                    'form': form,
+                    'provider_id': provider_id,
+                }
+                return render(request, 'provider_form.html', context)
+
+            uploaded_file = request.FILES['file_logo']
+            fs = FileSystemStorage()
+            logo_name = fs.save(uploaded_file.name, uploaded_file)
+            logo_url = os.path.join(settings.MEDIA_URL, logo_name)
+
+            provider = get_object_or_404(
+                Provider,
+                id=provider_id
+            )
+
+            provider.logo_url = logo_url
+            provider.save()
+
+            context = {
+                'form': form,
+                'provider_id': provider_id,
+                'provider_logo': logo_url,
+
+            }
+            return render(request, 'provider_form.html', context)
+        # Submit button is pressed
+        else:
+            form = ProviderForm(request.POST)
+            if form.is_valid():
+                new_provider = Provider.objects.filter(pk=provider_id).first()
+                provider_type = form.cleaned_data['provider_type']
+                telephone = form.cleaned_data['telephone']
+                email = form.cleaned_data['email']
+                physical_address_line_1 = (
+                    form.cleaned_data['physical_address_line_1'])
+                physical_address_line_2 = (
+                    form.cleaned_data['physical_address_line_2'])
+                physical_address_city = (
+                    form.cleaned_data['physical_address_city'])
+                postal_address = (
+                    form.cleaned_data['postal_address'])
+                admissions_contact_no = (
+                    form.cleaned_data['admissions_contact_no'])
+                new_provider.provider_type = provider_type
+                new_provider.telephone = telephone
+                new_provider.email = email
+                new_provider.physical_address_line_1 = physical_address_line_1
+                new_provider.physical_address_line_2 = physical_address_line_2
+                new_provider.physical_address_city = physical_address_city
+                new_provider.postal_address = postal_address
+                new_provider.admissions_contact_no = admissions_contact_no
+                new_provider.save()
+                campus_list = request.POST.getlist('campus_name')
+                number_of_campuses = len(campus_list)
+                try:
+                    with transaction.atomic():
+                        for idx in range(number_of_campuses):
+                            campus_name = campus_list[idx]
+                            Campus.objects.create(provider=new_provider,
```

Comment:
I'm not sure about the use case here. @pierrealixt you might want to comment on this too.
This code infer that every campus name given in a provider is a new campus.
Does `Campus.objects.get_or_create` will be better?

Reply:
This code is not part of the PR. It was written by @Christiaanvdm